app/javascript/components/timeline-options/timeline-chart.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { LineChart } from '@carbon/charts-react';
+import { timelineUiOptions } from './timeline-helper';
+
+const TimelineChart = ({ data, title, buildTableData }) => {
+  const chartRef = useRef(null);
+  const chartOnClick = ({ detail }) => {
+    buildTableData(detail.datum);
+  };
+
+  useEffect(() => {
+    chartRef.current.chart.services.events.addEventListener(
+      'scatter-click',
+      chartOnClick
+    );
+  }, [chartRef]);
+
+  // Unmount
+  useEffect(
+    () => () => {
+      if (chartRef.current) {
+        chartRef.current.chart.services.events.removeEventListener(
+          'scatter-click',
+          chartOnClick
+        );
+      }
+    },
+    []
+  );
+
+  return (
+    <div>
+      <LineChart className="line_charts" data={data} options={timelineUiOptions(title)} ref={chartRef} />
+    </div>
+  );
+};
+
+TimelineChart.propTypes = {
+  data: PropTypes.arrayOf(PropTypes.any).isRequired,
+  title: PropTypes.string,
+  buildTableData: PropTypes.func.isRequired,
```

Comment:
We can remove the `isRequired` for `buildTableData`, since we are using `defaultProps`